docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,495 @@
+
+Please read
+[Agreeing to the CLA](https://loopback.io/doc/en/contrib/code-contrib.html#agreeing-to-the-cla),
+click on this
+[link](https://cla.strongloop.com/agreements/strongloop/loopback-next), and sign
+it. This is done once per repository.
+
+The pull request has a checklist that must be filled out appropriately.
+
+![submit_pr_template_1.png](./imgs/submit_pr_template_1.png)
+
+Of particular importance are the `'npm test passes on your machine'` and
+`'Sign the CLA (Contributor License Agreement)'` checkboxes.
+
+#### 4. Check CI status
+
+Ensure that the continuous integration (CI) jobs associated with your pull
+request complete successfully.
+
+![submit_pr_final_commit_ci_jobs_pass_1.png](./imgs/submit_pr_final_commit_ci_jobs_pass_1.png)
+
+#### 5. PR Review Process
+
+The review process is iterative, so it is normal that `multiple commits` may be
+required until the pull request is finally accepted.
+
+{% include tip.html content="After a pull request is created, any additional commits that you push to your remote feature branch automatically get added to the pull request. There is no need to close the initial pull request and create a new one." %}
+
+Reactively [rebase](https://help.github.com/articles/about-git-rebase/) your
+forked repository against the upstream master branch to keep them in synch; if
+needed.
+
+#### Before your PR is merged by maintainer
+
+Before the PR is merged by a
+maintainer,[squash your commits](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)
+into one commit or a few commits; if needed. This step helps to keep the
+repository's commit history clean and concise.
+
+### Beginner Instructions
+
+Here is a complete tutorial on how to submit a pull request (PR) for LoopBack
+v4.
+
+#### 1. Fork the loopback-next repository
+
+In your browser, navigate to : <https://github.com/strongloop/loopback-next> .
+
+Create your own fork of the repository, by pressing the `Fork` link on the
+right-hand side .
+
+![submit_pr_fork_main_repo.png](./imgs/submit_pr_fork_main_repo.png)
+
+When the forking process is complete, the repository will show up as
+`{ your user id }/loopback-next`. In my case, it is `emonddr/loopback-next` .
+
+![submit_pr_my_forked_repo.png](./imgs/submit_pr_my_forked_repo.png)
+
+#### 2. Create the feature branch
+
+Notice your repo has a `master` branch already created (refer to bottom left
+corner of the picture above). It is commonplace to have this branch represent
+the latest, clean version of this repository's content. For the purposes of your
+PR, let's create a `feature` branch with a name indicative of your changes. In
+my case, it is `emonddr-doc-changes`.
+
+Click on the `Clone or download` button
+
+![submit_pr_create_feature_branch_1.png](./imgs/submit_pr_create_feature_branch_1.png)
+
+This brings up tiny dialog with different choices : Clone with SSH, Use HTTPS,
+Open in Desktop, or Download Zip.
+
+In my case, I will leave it as `Clone with SSH`, and click on the
```

Comment:
Can we leave a note here that `Clone with SSH` option requires you to have SSH set up on your local client machine? If they use HTTPS, I believe they would have to enter their GitHub credentials.

Reply:
hmm. I keep getting told that I have to trim information down. Not sure I want to start showing them the steps of using https. they can look it up.